File: capture_proofs.js
const { chromium } = require('playwright');
const path = require('path');

(async () => {
  console.log('🚀 Launching Playwright Chromium for E2E Visual Screenshots...');
  const browser = await chromium.launch({ headless: true });
  const page = await browser.newPage({ viewport: { width: 1440, height: 900 } });

  // 1. Capture B2B Workbench UI
  console.log('📸 Navigating to B2B Workbench: http://localhost:3005/...');
  await page.goto('http://localhost:3005/', { waitUntil: 'networkidle' });
  await page.waitForTimeout(3000);
  const workbenchPath = '/Users/pranay/.gemini/antigravity/brain/c076d50a-93f3-406d-85f7-41b771ca0650/e2e_workbench_live.png';
  await page.screenshot({ path: workbenchPath, fullPage: false });
  console.log(`✅ Saved Workbench Screenshot to ${workbenchPath}`);

  // 2. Capture Interactive Traveler Proposal UI
  console.log('📸 Navigating to Interactive Proposal: http://localhost:3005/proposals/test-proposal-123...');
  await page.goto('http://localhost:3005/proposals/test-proposal-123', { waitUntil: 'networkidle' });
  await page.waitForTimeout(3000);
  const proposalPath = '/Users/pranay/.gemini/antigravity/brain/c076d50a-93f3-406d-85f7-41b771ca0650/e2e_proposal_page_live.png';
  await page.screenshot({ path: proposalPath, fullPage: false });
  console.log(`✅ Saved Proposal Screenshot to ${proposalPath}`);

  await browser.close();
  console.log('🎉 Playwright E2E Screenshot Capture Completed Successfully!');
})();

File: tools/capture_proofs.py
"""
tools/capture_proofs.py — Automated screenshot capture for visual verification.
Captures live screenshots of /intake/fast, /proposals/prop_demo123, and /corporate/offsites.
"""

import asyncio
from pathlib import Path
from playwright.async_api import async_playwright

ARTIFACT_DIR = Path("/Users/pranay/.gemini/antigravity/brain/c076d50a-93f3-406d-85f7-41b771ca0650")

async def main():
    async with async_playwright() as p:
        browser = await p.chromium.launch(headless=True)
        context = await browser.new_context(viewport={"width": 1280, "height": 800})
        page = await context.new_page()

        # 1. Capture /intake/fast
        print("Navigating to http://localhost:3005/intake/fast...")
        await page.goto("http://localhost:3005/intake/fast", wait_until="networkidle")
        await page.screenshot(path=str(ARTIFACT_DIR / "proof_intake_fast.png"))
        print("Captured proof_intake_fast.png")

        # 2. Capture /proposals/prop_demo123
        print("Navigating to http://localhost:3005/proposals/prop_demo123...")
        await page.goto("http://localhost:3005/proposals/prop_demo123", wait_until="networkidle")
        await page.screenshot(path=str(ARTIFACT_DIR / "proof_proposal_teaser.png"))
        print("Captured proof_proposal_teaser.png")

        # 3. Capture /corporate/offsites
        print("Navigating to http://localhost:3005/corporate/offsites...")
        await page.goto("http://localhost:3005/corporate/offsites", wait_until="networkidle")
        await page.screenshot(path=str(ARTIFACT_DIR / "proof_corporate_offsites.png"))
        print("Captured proof_corporate_offsites.png")

        await browser.close()

if __name__ == "__main__":
    asyncio.run(main())
